### ISSUE TITLE
maintenance mode should be exposed for use

### ISSUE THREAD

**Issue by btm** [chef/chef-server#2089]
tldr; it would be convenient:

* `chef-server-ctl maintenance on|off` was added 
* the whitelist works so you can add 127.0.0.1 (maybe `chef-server-ctl maintenance` sub-command?) - if added, rename that to allow/block list.

--
The "503_mode" and "dark launch" features of yester-year still lurk and have some value. Parts of this are used in `chef-server-ctl reindex` to prevent clients from using the search endpoint while it is being reindexed. being able to put the server in maintenance mode before `chef-server-ctl reconfigure` would be helpful, especially on an upgrade where much could change and it could take some time.

You can manually enable 503_mode with:
```
REDIS_PASS=`chef-server-ctl show-secret redis_lb password`
/opt/opscode/embedded/bin/redis-cli -p 16379 -a $REDIS_PASS HSET dl_default 503_mode true
```

and disable 503_mode with:
```
REDIS_PASS=`chef-server-ctl show-secret redis_lb password`
/opt/opscode/embedded/bin/redis-cli -p 16379 -a $REDIS_PASS HDEL dl_default 503_mode
```

But `chef-server-ctl reindex -a` breaks when using `-a` for all orgs if you already have 503_mode on, because it needs to access the API to get the list of organizations. There was a whitelist functionality, but either it never worked right or over time the nginx lua configuration got broken.

https://github.com/chef/chef-server/blob/13.2.0/omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/scripts/config.lua.erb

`is_addr_whitelisted` and `is_addr_banned` always return false, and don't appear to be called from anywhere anyway

those values probably need to be fixed here before we could pass them from the chef-server.rb:

https://github.com/chef/chef-server/blob/13.2.0/omnibus/files/private-chef-cookbooks/private-chef/recipes/redis_lb.rb#L114-L115

`is_in_maint_mode_for_addr` appears to be used but does nothing with the address, just returns the maintenance mode state:

https://github.com/chef/chef-server/blob/13.2.0/omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/scripts/config.lua.erb#L142-L146

Unfortunately? that recipe also uses and then clears maintenance mode, so you can't set it before a reconfigure and have it still be set after. Although maybe something else would fail without whitelisting localhost.

It appears you can set `lb['xdl_defaults']['503_mode'] = true` in the chef-server.rb and keep 503_mode on, so kind of hacky but:
1. edit /etc/opscode/chef-server.rb and add `lb['xdl_defaults']['503_mode'] = true`
2. chef-server-ctl reconfigure
3. edit /etc/opscode/chef-server.rb and remove `lb['xdl_defaults']['503_mode'] = true`
